### ISSUE TITLE
Native macOS support (arm64/Apple Silicon, without Rosetta).

### ISSUE THREAD

**Issue by elvey** [OpenRefine/OpenRefine#6052]
OpenRefine v3.7.5 won't run natively/without Rosetta on M1 / M2 / M3 -powered Macs.  It's compiled for Intel.  It seems like some subparts are being compiled into Universal apps, but not the main app.  I haven't installed Rosetta for security reasons. I'd like to use OpenRefine relatively safely and run it natively.

**Comment by wetneb** [comment]
We probably need two MacOS packages: one for x64, one for aarch64.
The build times will be quite long because notarizing two DMGs will take a really long time.

At the moment, all packages are generated by the same job, sequentially. Perhaps it is time to reconsider that and generate them on different jobs. We'd be running two MacOS jobs generating one DMG each. The downside of that is that all jobs will probably have to download dependencies and compile the app independently, unless we figure out a sensible way to share the intermediate compilation state.

**Comment by tfmorris** [comment]
This describes how to build a homegrown universal JRE: https://incenp.org/notes/2023/universal-java-app-on-macos.html
but this should really be solved by the Java platform folks, so I think it's blocked by https://bugs.openjdk.org/browse/JDK-8310589

I'm not sure it makes a ton of sense to invest lots of time/effort in workarounds like multiple builds (which the user then needs to choose correctly) rather than just waiting for the ecosystem to mature.

**Comment by elvey** [comment]
Thanks. Re. avoiding duplicate effort:fair enough.  I do note that [JDK-8310589] has Priority: P4: minor loss of function where easy workaround is present, however. 
I see Oracle offers, for macOS, arm64:
JDK 20 binaries are free to use in production and free to redistribute, at no cost, under the [Oracle No-Fee Terms and Conditions](https://java.com/freeuselicense).
https://download.oracle.com/java/20/latest/jdk-20_macos-aarch64_bin.dmg 
Separately, https://jdk.java.net/20/ ALSO offers a macOS / AArch64 (AKA macOS / arm64) binary. 
Maybe I'm clueless, but it seems like one or both of these would be usable?   The incenp note seems out of date in this respect.

**Comment by wetneb** [comment]
Yes JDKs are available for both architectures - the problem is that there is not a single JDK which works for both of them, so we'd have to publish two separate packages for both architectures.

**Comment by mhkeller** [comment]
Any update on this. It would be nice to have this work natively on newer Macs.

**Comment by tfmorris** [comment]
As you can see by following the links, nothing has changed upstream, so no changes here either.

**Comment by M-Rick** [comment]
> Yes JDKs are available for both architectures - the problem is that there is not a single JDK which works for both of them, so we'd have to publish two separate packages to support both architectures.

And so, what's the matter? Have you seen how many softwares are available like this on macOS?

**Comment by thadguidry** [comment]
My vote would be for us to improve the situation for our users by publishing hardware specific OpenRefine releases where users have expressed an interest in us supporting their hardware.

@M-Rick is it your opinion that we just need to publish 2 releases for Mac hardware?  or is it 3?  and what naming convention or phrase would you call each of those OpenRefine Mac releases, in your opinion?

**Comment by ncarboni** [comment]
> My vote would be for us to improve the situation for our users by publishing hardware specific OpenRefine releases where users have expressed an interest in us supporting their hardware.
> 
> [@M-Rick](https://github.com/M-Rick) is it your opinion that we just need to publish 2 releases for Mac hardware? or is it 3? and what naming convention or phrase would you call each of those OpenRefine Mac releases, in your opinion?
> 
> * openrefine-mac-x64-3.9.0.dmg
> * openrefine-mac-aarch64-3.9.0.dmg
> * ?
> 
> [@wetneb](https://github.com/wetneb) Yes, we should just have many jobs actually (split parallel builds - GitHub allows us [20 concurrent jobs and on macOS hardware 5 concurrent jobs](https://docs.github.com/en/actions/administering-github-actions/usage-limits-billing-and-administration). So more than enough for our hardware support matrix we'd need)

In case of two versions are available, generally the naming for the one developed for the apple silicon architecture contains 'silicon' or 'arm64'. So, in term of clarity, these names would be better.

* openrefine-mac-x64-3.9.0.dmg
* openrefine-mac-arm64-3.9.0.dmg

But sincerely, aarch64 is also perfectly fine. 

Two versions are definitively enough!

**Comment by dragon-dxw** [comment]
This has become more urgent: [news](https://appleinsider.com/articles/26/06/12/how-and-when-macos-will-finally-stop-support-for-intel-apps) is that MacOS 27 (September 2026) will uninstall Rosetta and MacOS 28 (September 2027) will not support it. End of life for Intel is September 2029, so the good news is we'll only need one type of release from then.

**Comment by Pluralog** [comment]
Yes. OpenRefine is not running on current Apple Betas.

**Comment by andylaw** [comment]
It is ironic that the documentation for building OpenRefine packages says:

> The packages are currently optimized to run on Mac OS X which is the only platform capable of creating the packages for all three OS that we support.

when the build process seems incapable of building a package that will run on that platform in the near future

**Comment by thadguidry** [comment]
@andylaw Could you help fix the build and then improve the docs?

**Comment by andylaw** [comment]
@thadguidry I am very time-poor at the moment, but I'll be looking at it.

**Comment by andylaw** [comment]
I'm struggling to get the build and packaging process understood and working. There are multiple conflicting sets of instructions. Who is the main package-meister in the process. I see that @SoryRawyer appears quite pivotal in the 3.10 release discussion. Can anyone point me towards a definitive set of instructions and/or explanation?

**Comment by tfmorris** [comment]
The builds are done by Github Actions using this workflow https://github.com/OpenRefine/OpenRefine/blob/master/.github/workflows/snapshot_release.yml
but I'd like to reorganize the process per #7692. Since it looks like there's never going to be a universal JVM, it looks like we're stuck adding an additional build target/kit, but that should be straightforward, if resource intensive.

@andylaw if you can point to the confusing/conflicting build instructions -- or open an issue -- we can get them corrected

**Comment by tfmorris** [comment]
I've got a sketch of a multi-architecture build working, so we MAY be able to deliver this for 3.10.2, if the changes aren't too extensive and we can resolve #7894 which is a blocker.
